src/xpu/sycl/usm_memory_storage.hpp

```diff
@@ -69,26 +69,8 @@ class usm_memory_storage_t : public memory_storage_base_t {
     bool is_host_accessible() const override {
         if (engine()->kind() == engine_kind::cpu) return true; // optimism
 
-        /* FIXME: remove the W/A below when possible (fixed driver).
-         * Currently, shared USM suffers from synchronization issues between
-         * different devices. Specifically, the data migration from GPU to CPU
-         * may misbehave if accessed from multiple threads.
-         *
-         * One of the possible ways to work around the issue is to pretend that
-         * shared USM is not accessible on the host, hence map will allocate an
-         * extra memory and do a true copy of the data. This is what is done
-         * here: the shared USM marked as not-host-accessible.
-         *
-         * There is another possible work-around: touch the data on the host in
-         * the main thread, causing the "sequential" data migration. This W/A
-         * is a little bit more preferable as it doesn't allocate extra memory
-         * on the host. However it didn't work well for benchdnn, though worked
-         * perfectly fine for gtests. As we weren't able to find the cause of
-         * this behavior we went with the approach above. Hopefully, the driver
-         * will be fixed and we can get rid of W/A altogether. */
         return utils::one_of(usm_kind_, ::sycl::usm::alloc::host,
-                // ::sycl::usm::alloc::shared, // W/A (see above)
-                ::sycl::usm::alloc::unknown);
+                ::sycl::usm::alloc::shared, ::sycl::usm::alloc::unknown);
```

Comment:
Random spot. I see benchdnn mentions this workaround:

https://github.com/uxlfoundation/oneDNN/blob/1496c4a0539489ab49b7adffa6577fff578d4ecb/tests/benchdnn/dnnl_common.cpp#L1318-L1323

This benchdnn comment probably needs adjustments. At the same time, it's simpler to keep `mapped_mem_factor` as is - the default is USM device memory which needs this extra adjustment.

@rjoursler Just curious, how did you step into that? We should use USM device memory by default in oneDNN so USM shared memory is not regularly tested.